xla/service/elemental_ir_emitter_test.cc

```diff
@@ -101,6 +101,162 @@ ENTRY main {
   RunTest(hlo_text, {&lhs, &rhs});
 }
 
+inline std::string llvmValueToString(llvm::Value* v) {
+  std::string buffer;
+  llvm::raw_string_ostream rso(buffer);
+  v->print(rso);
+  rso.flush();
+  return buffer;
+}
+
+XLA_TEST_F(ElementalIrEmitterExecutionTest, EmitReducePrecisionIR_F16ToF8e5m2) {
+  llvm::LLVMContext llvm_context;
+  llvm::IRBuilder<> builder(llvm_context);
+  llvm::IRBuilder<>* b = &builder;
+  llvm::Type* f16_type = b->getHalfTy();
+
+  float inf = std::numeric_limits<float>::infinity();
+  float qnan = std::numeric_limits<float>::quiet_NaN();
+  float snan = std::numeric_limits<float>::signaling_NaN();
+
+  struct TestCase {
```

Comment:
can you split struct declaration from variable declaration, this is too exotic for XLA codebase

Reply:
This approach is used in 
- xla/tests/convert_test.cc (in 14 tests)
- xla/util_test.cc (in CommonFactors)

BTW, struct name "TestCase" can be omitted (example xla/util_test.cc CommonFactors)

Reply:
I agree with Eugene's comment. Despite this approach being used else where, it would better to split the declarations.

Reply:
Sure, I refactored the test file and added
```
struct EmitReducePrecisionIrTestCase {
  float input;
  std::string expected_res;
};
```

I also rebased the branch and switched to `llvm::IRBuilderBase* b`.

Reply:
Thanks!